contracts/LockDealNFT/LockDealNFT.sol

```diff
@@ -99,8 +99,9 @@ contract LockDealNFT is LockDealNFTModifiers, IERC721Receiver {
             }
         }
         if (withdrawnAmount > 0) {
-            emit MetadataUpdate(poolId);
             vaultManager.withdrawByVaultId(poolIdToVaultId[poolId], from, withdrawnAmount);
+            emit MetadataUpdate(poolId);
+            emit TokenWithdrawn(poolId, from, withdrawnAmount, getData(poolId).params[0]);
```

Comment:
all this can be in handle withdraw function, 101-104
```
vaultManager.withdrawByVaultId(poolIdToVaultId[poolId], from, withdrawnAmount);
            emit MetadataUpdate(poolId);
            emit TokenWithdrawn(poolId, from, withdrawnAmount, getData(poolId).params[0]);
            ```